tests/behaviour/typeql/steps.rs

```diff
@@ -21,48 +21,76 @@
-        // FIXME parse
-        let res = context.transaction().query().define(step.docstring().unwrap()).await;
-        assert!(res.is_err());
-        assert!(res.unwrap_err().to_string().contains(&exception));
+        match parse_query(step.docstring().unwrap()) {
+            Ok(parsed) => {
+                let res = context.transaction().query().define(&parsed.to_string()).await;
+                assert!(res.is_err());
+                assert!(res.unwrap_err().to_string().contains(&exception));
+            },
+            Err(error) => assert!(error.to_string().contains(&exception)),
+        }
     }
 
     #[step(expr = "typeql insert")]
     async fn typeql_insert(context: &mut Context, step: &Step) {
-        // FIXME parse
-        let stream = context.transaction().query().insert(step.docstring().unwrap()).unwrap();
-        stream.try_collect::<Vec<_>>().await.unwrap();
+        let parsed = parse_query(step.docstring().unwrap());
+        assert!(parsed.is_ok());
+        let stream = context.transaction().query().insert(&parsed.unwrap().to_string());
+        assert!(stream.is_ok());
+        let res = stream.unwrap().try_collect::<Vec<_>>().await;
+        assert!(res.is_ok());
     }
 
     #[step(expr = "typeql insert; throws exception")]
     async fn typeql_insert_throws(context: &mut Context, step: &Step) {
-        // FIXME parse
-        assert!(context.transaction().query().insert(step.docstring().unwrap()).is_err());
+        let parsed = parse_query(step.docstring().unwrap());
+        if parsed.is_ok() {
+            let stream = context.transaction().query().insert(&parsed.unwrap().to_string());
+            if stream.is_ok() {
+                let res = stream.unwrap().try_collect::<Vec<_>>().await;
+                assert!(res.is_err());
+            }
+        }
     }
 
     #[step(expr = "typeql insert; throws exception containing {string}")]
     async fn typeql_insert_throws_exception(context: &mut Context, step: &Step, exception: String) {
-        // FIXME parse
-        let res = context.transaction().query().insert(step.docstring().unwrap()).unwrap().try_collect::<Vec<_>>().await;
-        assert!(res.is_err());
-        assert!(res.map(|_| ()).unwrap_err().to_string().contains(&exception));
+        match parse_query(step.docstring().unwrap()) {
+            Ok(parsed) => {
+                match context.transaction().query().insert(&parsed.to_string()) {
+                    Ok(stream) => {
+                        let res = stream.try_collect::<Vec<_>>().await;
+                        assert!(res.is_err());
+                        assert!(res.map(|_| ()).unwrap_err().to_string().contains(&exception));
+                    },
+                    Err(error) => assert!(error.to_string().contains(&exception)),
+                }
+            },
+            Err(error) => assert!(error.to_string().contains(&exception)),
+        }
```

Comment:
It's a bit of a smell that we perform the same check here repeatedly. This can be streamlined. Look into `Result` combinators such as `and_then()`.